src/aws-manager.js

```diff
@@ -31,18 +89,32 @@ class AwsManager {
     assert(pubKey);
     assert(maxInstanceLife);
     assert(influx);
+
     this.ec2 = ec2;
     this.provisionerId = provisionerId;
     this.keyPrefix = keyPrefix;
     this.pubKey = pubKey;
     this.maxInstanceLife = maxInstanceLife;
     this.influx = influx;
+
+    // Known keypairs are tracked so that we don't have to retreive the list of
+    // all known key pairs on every iteration.
     this.__knownKeyPairs = [];
+
+    // The responses of the EC2 Api's view of state
     this.__apiState = {
       instances: [],
       requests: [],
     };
+
+    // We store the state of the previous iteration so that we can do
+    // comparisons to see which instances have changed state
     this.__previousApiState = this.__apiState;
+
+    // Internal state is state which we know exists but has not yet been
+    // reflected in the EC2 api.  This list is used to track the spot requests
+    // which we have submitted but which do not yet show up in the ec2 api
+    // calls which list the spot requests
     this.__internalState = [];
```

Comment:
I really wish this was called something more intuitive... Like `_submittedSpotRequests` though that is way too long... :)
Just saying...